### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by soerenlemke**
- Connection now checking if the database file is a valid database
- Connection status return values swapped for matching other functions
- tests fixed for swapped return values of ConnectionStatus

### CODE REVIEW

File: database/connection.go [hunk 1]
```diff
@@ -29,13 +32,11 @@ func (db *Database) Close() error {
 	return nil
 }
 
-func (db *Database) ConnectionStatus() (err error, ok bool) {
-	// Returns error if the file is not a valid database
-	
+func (db *Database) ConnectionStatus() (ok bool, err error) {
```
Comment (vyshnav-vinod) [inline database/connection.go:35]: Ok we can just make it return the error then. So if connection was fail, err will be set.. else err will be nil

**Review by vyshnav-vinod — CHANGES_REQUESTED** [review]

File: database/connection.go (hunk 1 above)
Comment (vyshnav-vinod) [reply database/connection.go:35]: I believe a ok and error together is unnecessary.

Comment (soerenlemke) [reply database/connection.go:35]: ok

**Comment by soerenlemke** [comment]
fixed it

**Review by vyshnav-vinod — APPROVED** [review]
Looks good

**Pull request merged**